docs/Bitbucket.md

```diff
@@ -0,0 +1,107 @@
+| `pipelines.default.step.services` | ✅ | The name of one or more services defined at `definitions.services.<name>` that will be applied for this step. Translated to utilise the service configuration with the [docker-compose-buildkite-plugin](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin). |
+| `pipelines.default.step.script` | ✅ | The individual commands that make up a specific step. Each is translated into a singular command within the `commands` key of a Buildkite command step. |
+| `pipelines.default.step.script.pipe` | ❌ | Re-usable modules to make configuration in Bitbucket pipelines easier and modular. Consider using [Buildkite Plugins](https://buildkite.com/docs/plugins) and/or [Pipeline Templates](https://buildkite.com/docs/pipelines/templates) in configuring re-usable configuration accross a Buildkite organization. |
+| `pipelines.default.step.size` | ✅ | Allocation of sizing options for the given memory for a specific step within a Bitbucket pipeline. The `size` value will be set as an agent [tag](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) in the Buildkite command step for targeting on specific Buildkite agents within an organization. |
+| `pipelines.default.step.trigger` | ✅ | The configuration setting the running of a Bitbucket pipeline step manually or automatically (latter being defaulted). For `manual` triggers - an [input step](https://buildkite.com/docs/pipelines/input-step) is inserted into the generated Buildkite pipeline before the specified `script` within a further command step. Explicit dependencies with `depends_on` are set between the two steps; requiring manual processing. |
+| `pipelines.default.variables` | ✅ | Custom variables that are passed to Bitbucket pipeline steps. Each variable defined in a Bitbucket pipeline step is translated to a Buildkite [input step](https://buildkite.com/docs/pipelines/input-step) with/without defaults and allowed values specified below. |
+| `pipelines.default.variables.name` | ✅ | The variables' name: translated to the `key` attribute of a specific `field` of an [input step](https://buildkite.com/docs/pipelines/input-step) (text entry).|
+| `pipelines.default.variables.default` | ✅ | The default variable value if no value is set. Set as the `default` attribute within the `field` of an [input step](https://buildkite.com/docs/pipelines/input-step). |
+| `pipelines.default.variables.description` | ✅ | The description of the variable. Translated to the `text` attribute of a specific `field` within the generated  [input step](https://buildkite.com/docs/pipelines/input-step). |
+| `pipelines.default.variables.allowed-values` | ✅ | The variable's allowed values: each option is translated to a singular `options` object with given value of a specific `field` of an [input step](https://buildkite.com/docs/pipelines/input-step). |
```

Comment:
I would move this out to its own section... specially because `default` pipelines do not allow for variables, only custom and exported pipelines (the one in definitions) do. Also there should be a warning as the code has a huge TODO to translate `$name` of defined variables to `buildkite-agent metadata get ...` calls to complete the functionality

Reply:
To come in the second pass for rearranging the `pipelines` section!

Reply:
Have just popped in the completely revitalised doc explaining starting conditions/pipeline properties (stage, step etc) separately with links between 👍